rust/connlib/tunnel/src/device_channel/tun_android.rs

```diff
@@ -100,16 +100,15 @@ impl ioctl::Request<GetInterfaceNamePayload> {
     }
 
     fn name(&self) -> std::borrow::Cow<'_, str> {
-        // Safety: The memory of `self.name` is always initialized.
-        let cstr = unsafe { std::ffi::CStr::from_ptr(self.name.as_ptr() as _) };
-
-        cstr.to_string_lossy()
+        String.from_utf8_lossy(&self.name)
     }
 }
 
 #[derive(Default)]
 #[repr(C)]
-struct GetInterfaceNamePayload;
+struct GetInterfaceNamePayload {
```

Comment:
I could very well be wrong, but I think what may have been happening here is that the `name` field was being optimized out due to a potential compiler bug for the 32-bit android targets.

So the `ioctl` writes the ifname into a squashed struct and `self.name` points to an invalid memory address, causing the `SEGV_MAPERR`.

Reply:
> So the `ioctl` writes the ifname into a squashed struct and `self.name` points to an invalid memory address, causing the `SEGV_MAPERR`.

The idea of using `repr(C)` is that the structs are indeed "squashed". In other words, it is important that `name` is defined first and then the fields of `payload` in the order they are defined as. `GetInterfaceMtuPayload` adds an additional `mtu` field to it  for example.